src/controlCenter/controlCenter.vala

```diff
@@ -207,6 +210,22 @@ namespace SwayNotificationCenter {
             });
         }
 
+        public void on_realize () {
+            // Set a minimum size for the window
+            int configuredWidth = int.max (ConfigModel.control_center_minimum_width, ConfigModel.instance.control_center_width);
+            int configuredHeight = int.max (ConfigModel.control_center_minimum_height, ConfigModel.instance.control_center_height);
+
+            // Clip the width and height to the display dimensions
+            // if they are too big
+            Gdk.Monitor monitor = get_display ().get_primary_monitor () ?? get_display ().get_monitor (0);
```

Comment:
This only seems to get the primary monitor and not the monitor which the Window is focused on

Reply:
Maybe we'll need a wlroots vaapi? Hopefully We'll be able to solve it without that but I'm not sure if it'll work with X11 disabled...

Reply:
Seems like `Gdk.Monitor monitor = this.get_display ().get_monitor_at_window (this.get_window ());` works **BUT** sometime after the map event is called. I'm not sure which event to listen to but `draw` and `check_resize` kinda work but get executed many times...